galasa-parent/dev.galasa.framework.api.users/src/test/java/dev/galasa/framework/api/users/internal/routes/UsersRouteTest.java

```diff
@@ -0,0 +1,207 @@
+import dev.galasa.framework.api.common.mocks.MockAuthStoreService;
+import dev.galasa.framework.api.common.mocks.MockEnvironment;
+import dev.galasa.framework.api.common.mocks.MockFramework;
+import dev.galasa.framework.api.common.mocks.MockFrontEndClient;
+import dev.galasa.framework.api.common.mocks.MockHttpServletRequest;
+import dev.galasa.framework.api.common.mocks.MockHttpServletResponse;
+import dev.galasa.framework.api.common.mocks.MockTimeService;
+import dev.galasa.framework.api.common.mocks.MockUser;
+import dev.galasa.framework.api.users.mocks.MockUsersServlet;
+import dev.galasa.framework.spi.auth.IFrontEndClient;
+import dev.galasa.framework.spi.utils.GalasaGson;
+
+
+public class UsersRouteTest extends BaseServletTest {
+
+    Map<String, String> headerMap = Map.of("Authorization", "Bearer " + BaseServletTest.DUMMY_JWT);
+
+    GalasaGson gson = new GalasaGson();
+    
+    @Test
+    public void testUsersGetRequestReturnsUserByLoginIdReturnsOK() throws Exception {
+        // Given...
+        MockEnvironment env = new MockEnvironment();
+        MockTimeService mockTimeService = new MockTimeService(Instant.now());
+        MockAuthStoreService authStoreService = new MockAuthStoreService(mockTimeService);
+        MockFramework framework = new MockFramework(authStoreService);
+
+        String baseUrl = "http://my.server/api";
+
+
+        Map<String, String[]> queryParams = new HashMap<>();
+
+        queryParams.put("loginId", new String[] { "user-1" });
+
+        env.setenv(EnvironmentVariables.GALASA_USERNAME_CLAIMS, "preferred_username");
+        env.setenv(EnvironmentVariables.GALASA_EXTERNAL_API_URL,baseUrl);
+        MockUsersServlet servlet = new MockUsersServlet(framework, env);
+
+
+        MockHttpServletRequest mockRequest = new MockHttpServletRequest(queryParams,null, headerMap);
+        MockHttpServletResponse servletResponse = new MockHttpServletResponse();
+        ServletOutputStream outStream = servletResponse.getOutputStream();
+
+        {
+            MockFrontEndClient newClient = new MockFrontEndClient("web-ui");
+            newClient.name = "web-ui";
+            newClient.lastLoginTime = Instant.parse("2024-10-18T14:49:50.096329Z");
+            
+
+            MockUser user1Data = new MockUser();
+            user1Data.userNumber = "docid";
+            user1Data.loginId = "user-1";
+            user1Data.addClient(newClient);
+
+            authStoreService.addUser(user1Data);
+        }
+
+        {
+            MockFrontEndClient newClient = new MockFrontEndClient("rest-api");
+            newClient.name = "rest-api";
+            newClient.lastLoginTime = Instant.parse("2024-10-18T14:49:50.096329Z");
+            
+
+            MockUser user1Data = new MockUser();
+            user1Data.userNumber = "docid-2";
+            user1Data.loginId = "user-2";
+            user1Data.addClient(newClient);
+
+            authStoreService.addUser(user1Data);
+        }
```

Comment:
Perhaps these blocks can be two function calls? Something like `createUser(String loginId, String userNumber, String clientName)`

Same applies for the test below